xaynetwork/xayn_discovery_app
# TB-4200: Fix bug which occurs after AI reset

### What 🕵️ 🔍

- Resetting AI implicates that the current feed and search are wiped on the engine side.
- We were still assuming documents were present (wrongly)

This PR adds:
- a reset AI handler on search (which was missing, we have a similar one for the feed)
- a new option, flag called `allowSuspension`, this can be set to false inside an observer.

The document time observer is now using `allowSuspension => false` and therefore stays "open" as we switch screens.

While not ideal, I couldn't think of a situation where this could cause issues.

----------

### How to test (please adjust template) 🥼 🔬
- Open feed and wait for load
- Open search and search anything
- Open settings and reset AI
- Revisit Search & Feed
- Expect no exception
